core/cas-server-core-web-api/src/main/java/org/apereo/cas/services/web/support/RegisteredServiceResponseHeadersEnforcementFilter.java

```diff
@@ -58,6 +58,8 @@ protected void decideInsertXFrameOptionsHeader(final HttpServletResponse httpSer
         final String xFrameOptions = getStringProperty(httpServletRequest, RegisteredServiceProperties.HTTP_HEADER_XFRAME_OPTIONS);
         if (xFrameOptions != null) {
             super.setXFrameOptions(xFrameOptions);
+        } else {
+            super.setXFrameOptions(RegisteredServiceProperties.HTTP_HEADER_XFRAME_OPTIONS.getDefaultValue());
```

Comment:
This won't work. You cant set the default based on what is defined in the code. The default (or the user-set value that is global) is retrieved from CAS settings in i.e. the cas.properties file. Because you overwrite it, you can't really reset it again back to what it was.

Ideally, you should not do `super.setXFrameOptions(xFrameOptions);` at all which changes the state of the parent filter. You need something like this:

```
super.insertWhatever(httpServletResponse, httpServletRequest, myCustomStuff);
```

And you should only do this, when there is indeed the need to set/insert that custom stuff, and not before that check (which is what the code does now). This is not the sort of thing you can do today, because the parent filter API does not let you do so. Once that is changed, I'll post a note here and you can try to refactor.

Reply:
Ah makes sense. Totally forgot about the possibility of overriding via cas.properties. Definitely the right thing to do to change the security filter implementation.